### PR TITLE
Update button state colors

### PR DESCRIPTION
**Pull request by chiuam**
### Platforms Impacted
- [ ] iOS
- [X] macOS

### Description of changes
Added new named colors for hover and pressed state colors and apply them in Button.swift

### Verification
Tested button state colors in test app.

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui-apple/pull/143)

### CODE REVIEW

File: macos/FluentUI/Button.swift [hunk 1]
```diff
@@ -193,6 +193,10 @@ open class Button: NSButton {
 
 	private var hoverBackgroundColor: NSColor {
 		if #available(macOS 10.14, *) {
+			if (fillColor == .clear)
+			{
```
Comment (anandrajeswaran) [inline macos/FluentUI/Button.swift:197]: swift (and therefore fluentui-apple) convention is same-line brackets like the rest of this file uses (same comment below). We really should get swift-format hooked up to make this automatic (was really hoping @Saadnajmi would do that this week :))

File: macos/FluentUIResources-macos/FluentUI.xcassets/ButtonView/Contents.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "info" : {
+    "author" : "xcode",
+    "version" : 1
+  },
+  "properties" : {
+    "provides-namespace" : true
```
Comment (anandrajeswaran) [inline macos/FluentUIResources-macos/FluentUI.xcassets/ButtonView/Contents.json:7]: Should the name of this namespace just be "Button" instead of "ButtonView" to match the control name?

File: macos/FluentUI/Button.swift (hunk 1 above)
Comment (anandrajeswaran) [reply macos/FluentUI/Button.swift:197]: Also looks like there's an extra space between "return" and "NSColor" here and below

File: macos/FluentUI/Button.swift [hunk 3]
```diff
@@ -193,6 +193,10 @@ open class Button: NSButton {
 
 	private var hoverBackgroundColor: NSColor {
 		if #available(macOS 10.14, *) {
+			if (fillColor == .clear)
```
Comment (anandrajeswaran) [inline macos/FluentUI/Button.swift:196]: Should this be in the #available(macOS 10.14, *) block? The code being added doesn't use anything 10.14 specific

File: macos/FluentUIResources-macos/FluentUI.xcassets/ButtonView/Contents.json (hunk 2 above)
Comment (HeyImChris) [reply macos/FluentUIResources-macos/FluentUI.xcassets/ButtonView/Contents.json:7]: We run into the same issue in FURN and I'd like to be consistent with naming whatever we decide. Personally I prefer dropping the word "view" as it better aligns with Apple control names and this is an Apple repo

File: macos/FluentUI/Button.swift (hunk 3 above)
Comment (Saadnajmi) [inline macos/FluentUI/Button.swift:196]: Would prefer wrapping the logic in a switch statement based on the ButtonStyle, rather than checking what the fillColor is

File: macos/FluentUI/Button.swift (hunk 1 above)
Comment (Saadnajmi) [reply macos/FluentUI/Button.swift:197]: Decided to add a clang format to the objective C code Amanda's team normally owns.

Tl;DR from my last FHL, it's easy to add a formatter as an optional thing, it's hard to include it as a dependency to this repo / enforce it is on every developers machine. I suppose at the very least we can add a swift-format file though...

Comment (Saadnajmi) [reply macos/FluentUI/Button.swift:197]: Also @anandrajeswaran I've got about 4 changes of varying size I'm doing this week so I highly doubt I'll get to add it :/

Comment (harrieshin) [reply macos/FluentUI/Button.swift:197]: if you run "swiftlint autocorrect" on the root of the repo iOS swift lint file works for mac code too. @chiuam  did you install swiftlint on your machine?

**Comment by harrieshin** [comment]
add screenshots please in the PR

**Comment by harrieshin** [comment]
also checkbox the testing of the PR

**Comment by chiuam** [comment]
Closing this PR as @chripear will be making these changes for Fluent vNext
